swap/tests/harness/mod.rs

```diff
@@ -43,13 +43,52 @@ const BITCOIN_TEST_WALLET_NAME: &str = "testwallet";
 #[derive(Debug, Clone)]
 pub struct StartingBalances {
     pub xmr: monero::Amount,
+    pub xmr_outputs: Vec<monero::Amount>,
     pub btc: bitcoin::Amount,
 }
 
+impl StartingBalances {
+    /// If monero_outputs is specified the monero balance will be:
+    /// monero_outputs * new_xmr = self_xmr
+    pub fn new(btc: bitcoin::Amount, xmr: monero::Amount, monero_outputs: Option<u64>) -> Self {
+        match monero_outputs {
+            None => {
+                if xmr == monero::Amount::ZERO {
+                    return Self {
+                        xmr,
+                        xmr_outputs: vec![],
+                        btc,
+                    };
+                }
+
+                Self {
+                    xmr,
+                    xmr_outputs: vec![xmr],
+                    btc,
+                }
+            }
+            Some(outputs) => {
+                let mut xmr_outputs = Vec::new();
+                let mut sum_xmr = monero::Amount::ZERO;
+
+                for _ in 0..outputs {
+                    xmr_outputs.push(xmr);
+                    sum_xmr = sum_xmr + xmr;
+                }
+
+                Self {
+                    xmr: sum_xmr,
+                    xmr_outputs,
+                    btc,
+                }
+            }
```

Comment:
I found it more readable from the outside when the amount it zero - otherwise one would have to pass an empy vec there.

Reply:
If you name the parameter `split_into_n_utxos` then it would always make sense to pass a value, the default would simply be 1.

Not too important though :)

Reply:
For zero it would be 0 and I find it more intuitive to have a `None` then. 
I don't have strong feelings about it though, feel free to change it if it bugs you :)